src/main/scala/viper/server/core/VerificationWorker.scala

```diff
@@ -0,0 +1,403 @@
+        Some(instance.asInstanceOf[SilFrontend])
+      case Some(instance) =>
+        throw ViperServerWrongTypeException(instance.getClass.getName)
+      case _ =>
+        throw ViperServerBackendNotFoundException(clazzName)
+    }
+  }
+
+  // Implementation of the Reporter interface used by the backend.
+  class ActorReporter(val tag: String) extends Reporter {
+    val name = s"ViperServer_$tag"
+
+    def report(msg: Message): Unit = {
+      enqueueMessages(SilverEnvelope(msg))
+    }
+  }
+
+  def run(): Unit = {
+    try {
+      command match {
+        case "silicon" :: args =>
+          logger.info("Creating new Silicon verification backend.")
+          backend = new ViperBackend(new SiliconFrontend(new ActorReporter("silicon"), logger), program)
+          backend.execute(args)
+        case "carbon" :: args =>
+          logger.info("Creating new Carbon verification backend.")
+          backend = new ViperBackend(new CarbonFrontend(new ActorReporter("carbon"), logger), program)
+          backend.execute(args)
+        case custom :: args =>
+          logger.info(s"Creating new verification backend based on class $custom.")
+          backend = new ViperBackend(resolveCustomBackend(custom, new ActorReporter(custom)).get, program)
+          backend.execute(args)
+        case args =>
+          logger.error("invalid arguments: ${args.toString}",
+            "You need to specify the verification backend, e.g., `silicon [args]`")
+      }
+    }catch {
+      case _: InterruptedException =>
+      case _: java.nio.channels.ClosedByInterruptException =>
+      case e: Throwable =>
+        q_actor ! ReporterProtocol.ServerReport(ExceptionReport(e))
+        logger.trace(s"Creation/Execution of the verification backend ${if (backend == null) "<null>" else backend.toString} resulted in exception.", e)
+    }finally {
+      try {
+        backend.stop()
+      }catch {
+        case e: Throwable =>
+          logger.trace(s"Stopping the verification backend ${if (backend == null) "<null>" else backend.toString} resulted in exception.", e)
+      }
+    }
+    if (backend != null) {
+      logger.info(s"The command `${command.mkString(" ")}` has been executed.")
+      q_actor ! TaskProtocol.FinalServerReport(true)
+    } else {
+      logger.error(s"The command `${command.mkString(" ")}` did not result in initialization of verification backend.")
+      q_actor ! TaskProtocol.FinalServerReport(false)
+    }
+  }
+}
+
+class ViperBackend(private val _frontend: SilFrontend, private val _ast: Program) {
+
+  def backendName: String = _frontend.verifier.getClass.getName
+  def file: String = _frontend.config.file()
+
+  override def toString: String = {
+    if ( _frontend.verifier == null )
+      s"ViperBackend( ${_frontend.getClass.getName} /with uninitialized verifier/ )"
+    else
+      s"ViperBackend( ${_frontend.verifier.name} )"
+  }
+
+  private def collectDefinitions(program: Program): List[Definition] = {
```

Comment:
[Minor] This method should be moved to Silver. IT shouldn't clutter the code here in ViperBackend.